src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4035,10 +4037,18 @@ private void ParseParameterModifiers(SyntaxListBuilder modifiers)
                 {
                     case SyntaxKind.ThisKeyword:
                         modifier = CheckFeatureAvailability(modifier, MessageID.IDS_FeatureExtensionMethod);
+                        if (seenRefModifier)
+                        {
+                            modifier = CheckFeatureAvailability(modifier, MessageID.IDS_FeatureRefExtensionMethods);
```

Comment:
When user uses a feature we complain that the feature is not available. 
Every version has multiple new features so this approach is not new.

We could just produce a generic "you are using 7.2 feature 'X' here ..." 
I do not have a strong opinion on this. I think it is ok either way.